src/Components/Search/SearchBar.tsx

```diff
@@ -116,79 +114,58 @@ export class SearchBar extends Component<Props, State> {
 
     if (entityType === "FirstItem") return
 
-    this.setState({
-      entityID,
-      entityType,
-    })
+    setEntityID(entityID)
+    setEntityType(entityType)
   }
 
-  @track((_props, _state, [query, hasResults]) => ({
+  track((_props, _state, [query, hasResults]) => ({
     action_type: hasResults
       ? DeprecatedSchema.ActionType.SearchedAutosuggestWithResults
       : DeprecatedSchema.ActionType.SearchedAutosuggestWithoutResults,
     query,
   }))
-  trackSearch(_term, _hasResults) {
+
+  const trackSearch = (_term, _hasResults) => {
     /* no-op */
   }
 
   // Throttled method to perform refetch for new suggest query.
-  throttledFetch = ({ value: term }) => {
-    const { relay } = this.props
-    const performanceStart = performance && performance.now()
-
-    relay.refetch(
-      {
-        hasTerm: true,
-        term,
-      },
-      null,
-      error => {
-        if (error) {
-          logger.error(error)
-          return
-        } else if (performanceStart && getENV("VOLLEY_ENDPOINT")) {
-          this.reportPerformanceMeasurement(performanceStart)
+  const throttledFetch = useCallback(
+    ({ value: term }) => {
+      const performanceStart = performance && performance.now()
+
+      relay.refetch(
+        {
+          hasTerm: true,
+          term,
+        },
+        null,
+        error => {
+          if (error) {
+            logger.error(error)
+            return
+          } else if (performanceStart && getENV("VOLLEY_ENDPOINT")) {
+            reportPerformanceMeasurement(performanceStart)
+          }
+          const edges = get(viewer, v => v.searchConnection.edges, [])
+          trackSearch(term, edges.length > 0)
         }
-        const { viewer } = this.props
-        // @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION
-        const edges = get(viewer, v => v.searchConnection.edges, [])
-        // @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION
-        this.trackSearch(term, edges.length > 0)
-      }
-    )
-  }
+      )
+    },
+    [relay, viewer]
+  )
 
-  componentDidMount() {
-    this.throttledFetch = throttle(this.throttledFetch, 500, {
+  useEffect(() => {
+    throttle(throttledFetch, 500, {
       leading: true,
     })
 
-    this.throttledOnSuggestionHighlighted = throttle(
-      this.throttledOnSuggestionHighlighted,
-      500,
-      { leading: true }
-    )
+    throttle(throttledOnSuggestionHighlighted, 500, { leading: true })
 
-    // Clear the search term once you navigate away from search results
-    this.removeNavigationListener = this.props.router
-      ? this.props.router.addNavigationListener(location => {
-          if (!location.pathname.startsWith("/search")) {
-            this.setState({ term: "" })
-          }
+    return removeNavigationListener
+  }, [removeNavigationListener, router, throttledFetch])
```

Comment:
you don't need to pass `throttledFetch` in deps for `useEffect`. Otherwise, the `useEffect` body will be called every time the `throttledFetch` changes (in the old implementation, this was called only at the mount stage)


https://user-images.githubusercontent.com/3513494/231861951-0d0b925d-b82c-4f76-8e4e-90dc55853690.mp4

Reply:
and should we pass `router` in deps?